crates/turborepo-lib/src/run/builder.rs

```diff
@@ -433,6 +460,12 @@ impl RunBuilder {
         }))
         .build()?;
 
+        // If we have an initial task, we prune out the engine to only
+        // tasks that are reachable from that initial task.
+        if let Some(entrypoint_tasks) = &self.entrypoint_tasks {
```

Comment:
This is a little hacky, but I couldn't figure out how to do this otherwise without cloning `Run` and we can't implement `Clone` on `Run` since we have the `AnalyticsHandle`.

Reply:
Are we rebuilding the original task graph each time?

Reply:
We might want to consider separating the `Engine` from an instance of `Run`. `Run` isn't really meant to be reusable, but it might take a little work to make it possible to use multiple of them.

Reply:
Yeah I think we need to work out what exactly is reusable and what is persistent across runs. Like we can reuse the same analytics handle and cache handle, but maybe not the same task graph